Metalama.Testing.AspectTesting/TestResult.cs

```diff
@@ -294,16 +305,48 @@ public IReadOnlyList<SyntaxTree> GetTestOutputsWithDiagnostics()
         // Adding the syntax of the transformed run-time code, but only if the pipeline was successful.
         var outputSyntaxTrees =
             this.TestInput.Options.OutputAllSyntaxTrees == true
-                ? this.SyntaxTrees.OrderBy( x => x.InputPath, StringComparer.InvariantCultureIgnoreCase ).AsEnumerable()
-                : this.SyntaxTrees.Take( 1 );
+                ? this.SyntaxTrees.OrderBy( x => x.InputPath ?? x.OutputDocument.AssertNotNull().FilePath, StringComparer.InvariantCultureIgnoreCase ).ToArray()
```

Comment:
The logic `x.InputPath ?? x.OutputDocument.AssertNotNull().FilePath` could be moved to the a new `TestSyntaxTree.FilePath` class.